File: src/Lessons/5-PrefixSums/5.2-CountingDivs.js
function solution(A, B, K) {
    //This functions gets the number of integers divisible by K between A and B
    //Seems a mathetical problem this one avoid loops as possible
    let divisibleIntegers = 0;

    if(A%K == 0){
        divisibleIntegers++;
        divisibleIntegers += Math.floor((B-A)/K);
    } else {
        let num = A + 1;
        let keepFinding = true;
        while(keepFinding){
            if(num%K == 0){
                divisibleIntegers++;
                divisibleIntegers += Math.floor((B-num)/K);
                break;
            }
            num++;
        }
    }
    return divisibleIntegers;
}

console.log(solution(5, 22, 2))
//expect 7

File: src/Lessons/5-PrefixSums/5.2-CountingDivs.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(solution(2, 9, 3))
}
func solution(A, B, K int) int {
	//Return integers between A and B that are divisible by K
	divisibleNumbers := 0
	if A%K == 0 {
		divisibleNumbers++
		divisibleNumbers += int(math.Floor(float64((B - A) / K)))
	} else {
		num := A + 1
		keepSearching := true
		for keepSearching == true {
			if num%K == 0 {
				divisibleNumbers++
				divisibleNumbers += int(math.Floor(float64((B - num) / K)))
				keepSearching = false
				break
			}
			num++
		}
	}
	return divisibleNumbers
}
